app/controllers/auth/sessions_controller.rb

```diff
@@ -111,6 +111,13 @@ def prompt_for_two_factor(user)
     render :two_factor
   end
 
+  def require_no_authentication
+    super
```

Comment:
What does the upstream method do? Does it make sense to `super` instead of just doing what it does without the thing we don't want?

Reply:
It does some more low-level checks to see if a user is logged in, and, if that's the case, redirects to `after_sign_in_path_for` and sets the flash message.

I thought about the two following alternatives:
- do that in `after_sign_in_path_for` instead, but doing anything effectful in there seems weird
- rewrite `require_no_authentication` entirely, but I'm a bit uncomfortable with that as it is lower-level Devise/warden stuff